packages/component-base/src/slot-observe-controller.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * @license
+ * Copyright (c) 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { SlotController } from './slot-controller.js';
+
+/**
+ * A controller that observes slotted element mutations, especially ID attribute
+ * and the text content, and fires an event to notify host element about those.
+ */
+export class SlotObserveController extends SlotController {
```

Comment:
I'm not sure about the name for new controller, as I might also try using it for `ErrorController` that shares some of the common logic, but does not need to observe mutations of the custom error message node.